diff --git a/arch/arm/boot/dts/imx6qdl-var-mx6cb-misc.dtsi b/arch/arm/boot/dts/imx6qdl-var-mx6cb-misc.dtsi
index 0d9fe55fcd1e..9fd79bb10387 100644
--- a/arch/arm/boot/dts/imx6qdl-var-mx6cb-misc.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-var-mx6cb-misc.dtsi
@@ -86,6 +86,7 @@
 		interrupt-parent = <&gpio5>;
 		interrupts = <17 IRQ_TYPE_EDGE_FALLING>;
 		touchscreen-max-pressure = <4096>;
+		touchscreen-inverted-x;
 		adi,resistance-plate-x = <250>;
 		adi,first-conversion-delay = /bits/ 8 <3>;
 		adi,acquisition-time = /bits/ 8 <1>;
