diff --git a/arch/arm/boot/dts/imx6qdl-var-mx6cb-cdisplay.dtsi b/arch/arm/boot/dts/imx6qdl-var-mx6cb-cdisplay.dtsi
index f66048eb7d2f..f7efdba8f6b3 100644
--- a/arch/arm/boot/dts/imx6qdl-var-mx6cb-cdisplay.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-var-mx6cb-cdisplay.dtsi
@@ -16,44 +16,45 @@
 	status = "okay";
 
 	lvds-channel@0 {
-		fsl,data-mapping = "spwg";
-		fsl,data-width = <24>;
+		fsl,data-mapping = "jeida";
+		fsl,data-width = <24>; 
 		status = "okay";
-		primary;
+
 
 		display-timings {
-			native-mode = <&timing0c>;
-			timing0c: hsd100pxn1 {
-				clock-frequency = <32000000>;
+			native-mode = <&timing0>;
+			timing0: hsd100pxn1 {
+				clock-frequency = <40000000>;
 				hactive = <800>;
-				vactive = <480>;
-				hback-porch = <39>;
-				hfront-porch = <39>;
-				vback-porch = <29>;
-				vfront-porch = <13>;
-				hsync-len = <47>;
-				vsync-len = <2>;
+				vactive = <600>;
+				hback-porch = <46>;
+				hfront-porch = <16>;
+				vback-porch = <23>;
+				vfront-porch = <12>;
+				hsync-len = <40>;
+				vsync-len = <20>;
 			};
 		};
 	};
 
 	lvds-channel@1 {
-		fsl,data-mapping = "spwg";
-		fsl,data-width = <24>;
-		status = "okay";
-
+		fsl,data-mapping = "jeida";
+		fsl,data-width = <24>;		
+		status = "okay";		
+		primary;
+		
 		display-timings {
 			native-mode = <&timing1>;
 			timing1: hsd100pxn1 {
-				clock-frequency = <38251000>;
+				clock-frequency = <32000000>;
 				hactive = <800>;
 				vactive = <600>;
-				hback-porch = <112>;
-				hfront-porch = <32>;
-				vback-porch = <3>;
-				vfront-porch = <17>;
-				hsync-len = <80>;
-				vsync-len = <4>;
+				hback-porch = <46>;
+				hfront-porch = <16>;
+				vback-porch = <23>;
+				vfront-porch = <12>;
+				hsync-len = <40>;
+				vsync-len = <20>;
 			};
 		};
 	};
diff --git a/arch/arm/boot/dts/imx6qdl-var-mx6cb-misc.dtsi b/arch/arm/boot/dts/imx6qdl-var-mx6cb-misc.dtsi
index 51cb8f4944b6..8c91387b072f 100644
--- a/arch/arm/boot/dts/imx6qdl-var-mx6cb-misc.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-var-mx6cb-misc.dtsi
@@ -63,6 +63,13 @@
 		compatible = "fsl,imx6-hdmi-i2c";
 		reg = <0x50>;
 	};
+
+	/* isl1208 RTC */
+	isl1208: rtc@6f {
+		compatible = "isil,isl1208";
+		reg = <0x6f>;
+		
+	};
 };
 
 &i2c3 {
@@ -71,6 +78,22 @@
 	pinctrl-0 = <&pinctrl_i2c3>;
 	status = "okay";
 
+	/* Touch */
+  res_touch: ad7879@2c {
+		compatible = "adi,ad7879-1";
+		reg = <0x2c>;
+		pinctrl-names = "default";
+		interrupt-parent = <&gpio5>;
+		interrupts = <17 IRQ_TYPE_EDGE_FALLING>;
+		touchscreen-max-pressure = <4096>;
+		adi,resistance-plate-x = <250>;
+		adi,first-conversion-delay = /bits/ 8 <3>;
+		adi,acquisition-time = /bits/ 8 <1>;
+		adi,median-filter-size = /bits/ 8 <3>;
+		adi,averaging = /bits/ 8 <1>;
+		adi,conversion-interval = /bits/ 8 <255>;					
+  };
+
 	/* Touch */
 	cap_touch: ft5x06_ts@38 {
 		compatible = "edt,edt-ft5x06";
@@ -81,12 +104,6 @@
 		interrupts = <7 0>;
 		status = "disabled";
 	};
-
-	/* DS1337 RTC module */
-	rtc@68 {
-		compatible = "dallas,ds1337";
-		reg = <0x68>;
-	};
 };
 
 &iomuxc {
@@ -182,7 +199,15 @@
 	status = "okay";
 };
 
-/* ttymxc2 UART */
+/* ttymxc1 UART slave*/
+&uart2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart2>;
+	uart-has-rtscts;
+	status = "okay";
+};
+
+/* ttymxc2 UART  Printer*/
 &uart3 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart3>;
diff --git a/arch/arm/boot/dts/imx6qdl-var-mx6cb-rdisplay.dtsi b/arch/arm/boot/dts/imx6qdl-var-mx6cb-rdisplay.dtsi
index 1e0002ad24b4..31439405d7a3 100644
--- a/arch/arm/boot/dts/imx6qdl-var-mx6cb-rdisplay.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-var-mx6cb-rdisplay.dtsi
@@ -16,44 +16,45 @@
 	status = "okay";
 
 	lvds-channel@0 {
-		fsl,data-mapping = "spwg";
-		fsl,data-width = <24>;
+		fsl,data-mapping = "jeida";
+		fsl,data-width = <24>; 
 		status = "okay";
-		primary;
+
 
 		display-timings {
-			native-mode = <&timing0r>;
-			timing0r: hsd100pxn1 {
-				clock-frequency = <30000000>;
+			native-mode = <&timing0>;
+			timing0: hsd100pxn1 {
+				clock-frequency = <40000000>;
 				hactive = <800>;
-				vactive = <480>;
-				hback-porch = <28>;
-				hfront-porch = <17>;
-				vback-porch = <13>;
-				vfront-porch = <20>;
-				hsync-len = <20>;
-				vsync-len = <13>;
+				vactive = <600>;
+				hback-porch = <46>;
+				hfront-porch = <16>;
+				vback-porch = <23>;
+				vfront-porch = <12>;
+				hsync-len = <40>;
+				vsync-len = <20>;
 			};
 		};
 	};
 
 	lvds-channel@1 {
-		fsl,data-mapping = "spwg";
-		fsl,data-width = <24>;
-		status = "okay";
-
+		fsl,data-mapping = "jeida";
+		fsl,data-width = <24>;		
+		status = "okay";		
+		primary;
+		
 		display-timings {
 			native-mode = <&timing1>;
 			timing1: hsd100pxn1 {
-				clock-frequency = <38251000>;
+				clock-frequency = <32000000>;
 				hactive = <800>;
 				vactive = <600>;
-				hback-porch = <112>;
-				hfront-porch = <32>;
-				vback-porch = <3>;
-				vfront-porch = <17>;
-				hsync-len = <80>;
-				vsync-len = <4>;
+				hback-porch = <46>;
+				hfront-porch = <16>;
+				vback-porch = <23>;
+				vfront-porch = <12>;
+				hsync-len = <40>;
+				vsync-len = <20>;
 			};
 		};
 	};
diff --git a/arch/arm/boot/dts/imx6qdl-var-som.dtsi b/arch/arm/boot/dts/imx6qdl-var-som.dtsi
index 44de570a173b..62616b5cccee 100644
--- a/arch/arm/boot/dts/imx6qdl-var-som.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-var-som.dtsi
@@ -235,7 +235,7 @@
 	status = "okay";
 
 	/* Touch */
-	res_touch: ads7846@0 {
+	res_touch_SPI: ads7846@0 {
 		reg = <0>;	/* CS0 */
 		compatible = "ti,ads7846";
 		pinctrl-names = "default";
@@ -580,11 +580,9 @@
 
 		/* Variscite Bluetooth */
 		pinctrl_uart2: uart2grp { /* RTS/CTS only mode */
-			fsl,pins = <
-				MX6QDL_PAD_SD3_DAT4__UART2_RX_DATA	0x1b0b1
-				MX6QDL_PAD_SD3_DAT5__UART2_TX_DATA	0x1b0b1
-				MX6QDL_PAD_EIM_D28__UART2_CTS_B		0x1b0b1
-				MX6QDL_PAD_EIM_D29__UART2_RTS_B		0x1b0b1
+			fsl,pins = <				
+				MX6QDL_PAD_EIM_D26__UART2_TX_DATA	  0x1b0b1
+				MX6QDL_PAD_EIM_D27__UART2_RX_DATA 	0x1b0b1				
 			>;
 		};
 
@@ -684,7 +682,7 @@
 	status = "okay";
 };
 
-/* Bluetooth Uart */
+/* Slave Uart */
 &uart2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart2>;
